pkg/network/node/ovscontroller.go

```diff
@@ -923,3 +935,17 @@ func (oc *ovsController) SetNamespaceEgressViaEgressIPs(vnid uint32, egressIPsMe
 	}
 	return otx.Commit()
 }
+
+func ensureLocalMAC(localIP string) string {
```

Comment:
We already check the primary NIC in `pkg/network/node.go:validateNodeIP()`, which calls `GetLinkDetails()` but ignores the result other than whether it returns an error. But you could just capture the MAC there as well.